Modules/_sqlite/module.c

```diff
@@ -416,9 +417,7 @@ PyMODINIT_FUNC PyInit__sqlite3(void)
        non-ASCII data and bytestrings to be returned for ASCII data.
        Now OptimizedUnicode is an alias for str, so it has no
        effect. */
-    Py_INCREF((PyObject*)&PyUnicode_Type);
-    if (PyModule_AddObject(module, "OptimizedUnicode", (PyObject*)&PyUnicode_Type) < 0) {
-        Py_DECREF((PyObject*)&PyUnicode_Type);
+    if (PyModule_AddObjectRef(module, "OptimizedUnicode", (PyObject*)&PyUnicode_Type) < 0) {
```

Comment:
Note unrelated to the PR: maybe it's time to consider removing this ghost from the past? I'm not sure how to introduce a deprecation warning. Maybe start in the documentation? ... I cannot find OptimizedUnicode in https://docs.python.org/dev/library/sqlite3.html but it is mentioned in the 2.7 doc: https://docs.python.org/2.7/library/sqlite3.html

Reply:
Yes, I've noticed this thing while digging around in the sqlite3 module. This isn't the only blast from the past in the sqlite3 module (look `sqlite3.enable_shared_cache()` for example.) I'll add an issue for it and try searching for similar cases in the mailing list archives / git log.

Reply:
Could it be possible to create a wrapper around `PyUnicode_Type` that emits a deprecation warning when used?

Reply:
I was thinking at a module __getattr__() which emits a DeprecationWarning at the first access. It is easy to do in Python, but I don't know how to do that in Python. But since it's not documented, it sounds safe to remove it with no deprecation period.

Reply:
I was thinking something like this: https://github.com/erlend-aasland/cpython/tree/sqlite-deprecate-optimized-unicode

Reply:
Actually, I would not add the wrapper to Lib/sqlite3/dbapi2.py, but rather create a new file, Lib/sqlite3/deprecated.py, and include that in __init__.py.

Reply:
> But since it's not documented, it sounds safe to remove it with no deprecation period.

I don't know; people seem to use undocumented stuff from time to time. Adding a deprecation costs little and it's friendly to the (few?) users of OptimizedUnicode :)

Reply:
FYI: https://bugs.python.org/issue13921

Reply:
Issue created: bpo-42264